### ISSUE TITLE
attribute ‘localtime’ missing (cron service)

### ISSUE THREAD

**Issue by fadenb** [NixOS/nixpkgs#28116]
Hey :)
## Issue description
After updating my nixpkgs to a more current version I am unable to deploy a system via NixOps:
```
λ becky nixos.wiki → λ git master* → nixops deploy -d wiki --copy-only                                                            ~/documents/code/nixos.wiki                           
building all machine configurations...
trace: warning: The services.postfix.extraConfig option was deprecated. Please use services.postfix.config instead.
error: attribute ‘localtime’ missing, at /home/fadenb/nixpkgs/nixos/modules/services/scheduling/cron.nix:125:31
(use ‘--show-trace’ to show detailed location information)
error: unable to build all machine configurations
```
It seems to be related to a0d464033c2bacffa9c05d27903fb03cd09ace46 (4f901203e8d1a0ded90a73e2670ce493ed051299).

I did not have `time.timeZone` configured. Once I set `time.timeZone = "UTC";` everything works as expected.

The documentation stated
`This is additionally now the default value for the option, which means that systems where the option has not been set explicitly will continue to use UTC but allow imperative timezone adjustments now.`
but has been changed to
`This value allows changing the timezone of a system imperatively using <command>timedatectl set-timezone</command>. The default timezone is still UTC.`.

Both variants do not imply I need to configure the timezone as it defaults to UTC. I believe this is a bug.

/cc @lheckemann

**Comment by lheckemann** [comment]
Yep looks like my fault. On it!

**Comment by lheckemann** [comment]
So this should be fixed by 85cab748c7.

However, this does not cause cron to restart when the timezone is changed imperatively so it doesn't fully solve the issue. I'll look into what can be done, but I'm not sure systemd supports taking additional actions on timezone change. Wrapping `timedatectl` could work, but it's really ugly IMHO. Not supporting this use case (i.e. "users have to restart cron manually if they change the timezone imperatively, or just use systemd timers instead of cron") is not optimal either, but it's the best solution I can think of. Any opinions are very welcome!

**Comment by lheckemann** [comment]
Could you edit the issue title to mention cron?

I've just found out that we can probably use a [.path unit](https://www.freedesktop.org/software/systemd/man/systemd.path.html) to watch /etc/localtime and restart services as appropriate.

**Comment by lheckemann** [comment]
I believe this is fixed by #28185.
